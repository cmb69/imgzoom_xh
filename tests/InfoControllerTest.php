<?php

namespace Imgzoom;

use ApprovalTests\Approvals;
use PHPUnit\Framework\TestCase;
use Plib\FakeRequest;
use Plib\FakeSystemChecker;
use Plib\View;

class InfoControllerTest extends TestCase
{
    public function testShowsPluginInfo(): void
    {
        $sut = new InfoController(
            "./",
            new FakeSystemChecker(),
            new View("./views/", XH_includeVar("./languages/en.php", "plugin_tx")["imgzoom"])
        );
        Approvals::verifyHtml($sut->defaultAction(new FakeRequest())->output());
    }
}
