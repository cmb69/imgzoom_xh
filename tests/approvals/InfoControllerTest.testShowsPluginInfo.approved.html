
<h1>Imgzoom 1.0</h1>
<div>
  <h2>System check</h2>
  <p class="xh_fail">Checking that the PHP version is at least 7.1.0 … failure</p>
  <p class="xh_fail">Checking that the CMSimple_XH version is at least 1.7.0 … failure</p>
  <p class="xh_fail">Checking that the CMSimple_XH plugin 'Plib_XH' is installed … failure</p>
  <p class="xh_warning">Checking that './css/' is writable … warning</p>
  <p class="xh_warning">Checking that './languages/' is writable … warning</p>
</div>
